From 1282f7e0cc939d08173e7bac8313127795ce853b Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:33 -0400
Subject: [tools] perf: Use machine->pid for tgid if machine is guest

Message-id: <1408953210-26343-206-git-send-email-jolsa@redhat.com>
Patchwork-id: 88962
O-Subject: [PATCH RHEL7.1 BZ1133083 205/322] perf tools: Use machine->pid for tgid if machine is guest.
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f5db57c4c47f3985dfc677152e25636a3401fa13
Author: Dongsheng Yang <yangds.fnst@cn.fujitsu.com>
Date: Fri Dec 20 15:52:59 2013 -0500

description
===========
When we synthesize an comm event, if machine is guest, we should
use the pid of machine as the event->comm.pid, rather than tgid
of thread.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/event.c b/tools/perf/util/event.c
index 07c0783..2905771 100644
--- a/tools/perf/util/event.c
+++ b/tools/perf/util/event.c
@@ -106,8 +106,12 @@ static pid_t perf_event__synthesize_comm(struct perf_tool *tool,
 
  memset(&event->comm, 0, sizeof(event->comm));
 
- tgid = perf_event__get_comm_tgid(pid, event->comm.comm,
-      sizeof(event->comm.comm));
+ if (machine__is_host(machine))
+  tgid = perf_event__get_comm_tgid(pid, event->comm.comm,
+       sizeof(event->comm.comm));
+ else
+  tgid = machine->pid;
+
  if (tgid < 0)
   goto out;
 
-- 
1.7.1